src/services/logger.js

```diff
@@ -39,8 +45,22 @@ class Logger {
     }
   }
 
-  _logLines(messages, options) {
-    messages.forEach((message) => this._logLine(message, options));
+  _logLines(messagesWithPotentialGivenOption, baseOptions) {
+    const { options, messages } = Logger._extractGivenOptionFromMessages(
+      messagesWithPotentialGivenOption,
+    );
+    messages.forEach((message) => this._logLine(message, { ...baseOptions, ...options }));
+  }
+
+  // this method is a hack to keep the signature of all existing public methods.
+  static _extractGivenOptionFromMessages(messages) {
+    let options = {};
+    const potentialGivenOptionFromCommand = messages[messages.length - 1];
```

Comment:
This is too permissive.
Any object given as the last parameter will be considered as an option object.
In your case, checking there is only one key ('lineColor`) should be done.